src/complete.ts

```diff
@@ -36,16 +36,44 @@ function sourceContext(state: EditorState, startPos: number) {
   if (pos.name == "Identifier" || pos.name == "QuotedIdentifier") {
     return {from: pos.from,
             quoted: pos.name == "QuotedIdentifier" ? state.sliceDoc(pos.from, pos.from + 1) : null,
-            parents: parentsFor(state, tokenBefore(pos))}
+            parents: resolveAlias(state, pos, parentsFor(state, tokenBefore(pos)))}
   } if (pos.name == ".") {
     return {from: startPos,
             quoted: null,
-            parents: parentsFor(state, pos)}
+            parents: resolveAlias(state, pos, parentsFor(state, pos))}
   } else {
     return {from: startPos, quoted: null, parents: [], empty: true}
   }
 }
 
+function resolveAlias(state: EditorState, node: SyntaxNode, parents: string[]): string[] {
+  if (parents.length === 1) {
+    const aliasName = parents[0];
+    // node.parent because the alias is referenced within a child (clause)
+    for (let searchNode = findFromClause(state, node.parent)?.nextSibling; searchNode != null && !isKeyword(state, searchNode, "WHERE"); searchNode = searchNode.nextSibling) {
+      if ((searchNode.name == "Identifier" || searchNode.name == "QuotedIdentifier") && stripQuotes(state.sliceDoc(searchNode.from, searchNode.to)) === aliasName) {
+        let sourceNode = isKeyword(state, searchNode.prevSibling, "AS") ? searchNode.prevSibling.prevSibling : searchNode.prevSibling;
+        if (sourceNode?.name.endsWith("Identifier")) { // can be Identifier, QuotedIdentifier or CompositeIdentifier
```

Comment:
We're targeting browsers that don't have `endsWith`—could you change this to `/Identifier$/.test`?

Reply:
Sure